modules/sdk-coin-ada/src/utils.ts

```diff
@@ -1,35 +1,90 @@
-import { BaseUtils, NotImplementedError } from '@bitgo/sdk-core';
+import { BaseUtils } from '@bitgo/sdk-core';
+import CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
+import bs58 from 'bs58';
+import { KeyPair } from './keyPair';
 
 export class Utils implements BaseUtils {
-  /** @inheritdoc */
+  validateBlake2b(hash: string): boolean {
+    if (hash.length !== 64) {
+      return false;
+    }
+    return hash.match(/^[a-zA-Z0-9]+$/) !== null;
+  }
 
+  /** @inheritdoc */
   isValidAddress(address: string): boolean {
-    throw new NotImplementedError('method not implemented');
+    try {
+      CardanoWasm.Address.from_bech32(address);
+      return true;
+    } catch (err) {
+      return false;
+    }
   }
 
   /** @inheritdoc */
   isValidBlockId(hash: string): boolean {
-    throw new NotImplementedError('method not implemented');
+    return this.validateBlake2b(hash);
   }
 
   /** @inheritdoc */
   isValidPrivateKey(key: string): boolean {
-    throw new NotImplementedError('method not implemented');
+    // this will return true for both extended and non-extended ED25519 keys
+    return this.isValidKey(key);
+  }
+
+  isBase58(value: string, length: number): boolean {
+    try {
+      return !!value && bs58.decode(value).length === length;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  base58Encode(value: Uint8Array): string {
+    return bs58.encode(value);
   }
 
   /** @inheritdoc */
-  isValidPublicKey(key: string): boolean {
-    throw new NotImplementedError('method not implemented');
+  isValidPublicKey(pubKey: string): boolean {
+    return this.isValidAccountId(pubKey);
```

Comment:
this should just check if can construct a key pair from public key, cardano is not account based.